src/main/kotlin/org/jitsi/jibri/api/http/HttpApi.kt

```diff
@@ -0,0 +1,108 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.jitsi.jibri.api.http
+
+import org.jitsi.jibri.CallParams
+import org.jitsi.jibri.FileRecordingParams
+import org.jitsi.jibri.JibriManager
+import org.jitsi.jibri.RecordingSinkType
+import org.jitsi.jibri.ServiceParams
+import org.jitsi.jibri.StartServiceResult
+import org.jitsi.jibri.StreamingParams
+import org.jitsi.jibri.util.extensions.debug
+import java.util.logging.Logger
+import javax.ws.rs.Consumes
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+// TODO: this needs to include usageTimeout
+data class StartServiceParams(
+    val callParams: CallParams,
+    val sinkType: RecordingSinkType,
+    val youTubeStreamKey: String? = null
+)
+
+/**
+ * The [HttpApi] is for starting and stopping the various Jibri services via the
+ * [JibriManager], as well as retrieving the health and status of this Jibri
+ */
+@Path("/jibri/api/v1.0")
+class HttpApi(private val jibriManager: JibriManager) {
+    private val logger = Logger.getLogger(this::class.qualifiedName)
+
+    /**
+     * Get the health of this Jibri in the format of a json-encoded
+     * [org.jitsi.jibri.health.JibriHealth] object
+     */
+    @GET
+    @Path("health")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun health(): Response {
+        logger.debug("Got health request")
+        return Response.ok(jibriManager.healthCheck()).build()
+    }
+
+    /**
+     * [startService] will start a new service using the given [StartServiceParams].
+     * Returns a response with [Response.Status.OK] on success, [Response.Status.PRECONDITION_FAILED]
+     * if this Jibri is already busy and [Response.Status.INTERNAL_SERVER_ERROR] on error
+     */
+    @POST
+    @Path("startService")
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun startService(serviceParams: StartServiceParams): Response {
+        logger.debug("Got a start service request with params $serviceParams")
+        val result: StartServiceResult = when (serviceParams.sinkType) {
+            RecordingSinkType.FILE -> run {
+                jibriManager.startFileRecording(
+                    ServiceParams(usageTimeoutMinutes = 0),
+                    FileRecordingParams(serviceParams.callParams),
+                    environmentContext = null
+                )
+            }
+            RecordingSinkType.STREAM -> run {
+                val youTubeStreamKey = serviceParams.youTubeStreamKey ?: return@run StartServiceResult.ERROR
```

Comment:
Nit: the val is unused, perhaps you want to use it on the line below instead of accessing serviceParams again?

Reply:
yup, thanks.  annoying that linter didn't catch this.  i'll have to look into that.